### ISSUE TITLE
[BUG] salt-ssh cannot execute module if the distant machine has only Python 3.5

### ISSUE THREAD

**Issue by HutTwoThreeFour** [saltstack/salt#66965]
**Description**
salt-ssh cannot run module on Ubuntu 16.04 machine with Python 3.5.

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)

Please be as specific as possible and give set-up details.

- [x] on-prem machine
- [x] VM (Hyper-V)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [ ] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
Running command `salt-ssh 'machine' test.ping` gives this error:

```
machine:
    ----------
    _error:
        The command resulted in a non-zero exit code
    parsed:
        None
    retcode:
        1
    stderr:
        Traceback (most recent call last):
          File "/var/tmp/.root_27c828_salt/salt-call", line 26, in <module>
            from salt.scripts import salt_call
          File "/var/tmp/.root_27c828_salt/pyall/salt/__init__.py", line 125, in <module>
            import salt._logging  # isort:skip
          File "/var/tmp/.root_27c828_salt/pyall/salt/_logging/__init__.py", line 13, in <module>
            from salt._logging.impl import (
          File "/var/tmp/.root_27c828_salt/pyall/salt/_logging/impl.py", line 112
            self.bracketname = f"[{str(self.name):<17}]"
                                                                               ^
        SyntaxError: invalid syntax
    stdout:

```

**Expected behavior**
Expected to have this result:
```
machine:
    True
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3007.1
 
Python Version:
        Python: 3.10.14 (main, Apr  3 2024, 21:30:09) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.16.0
      cherrypy: unknown
      dateutil: 2.8.2
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.4
       libgit2: Not Installed
  looseversion: 1.3.0
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.7
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 23.1
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.19.1
        pygit2: Not Installed
  python-gnupg: 0.5.2
        PyYAML: 6.0.1
         PyZMQ: 25.1.2
        relenv: 0.16.0
         smmap: Not Installed
       timelib: 0.3.0
       Tornado: 6.3.3
           ZMQ: 4.3.4
 
Salt Package Information:
  Package Type: onedir
 
System Versions:
          dist: ubuntu 22.04.2 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-106-generic
        system: Linux
       version: Ubuntu 22.04.2 jammy
```
</details>

**Additional context**
I guess it has something to do with Python version because I can run this line in Python 3.10 without problem:
`self.bracketname = f"[{str(self.name):<17}]"`
But if I run the same line in Python 3.5, it gives out the same error as when I run salt-ssh

**Comment by mastrong** [comment]
I don't think this can be considered a "bug". The lowest version of python supported is 3.7 https://docs.saltproject.io/salt/install-guide/en/3006/topics/salt-python-version-support.html#salt-python-version-support

**Comment by lkubb** [comment]
I think there is a system of using alternative dependencies for this case, but have never used it: https://docs.saltproject.io/en/latest/topics/ssh/ssh_ext_alternatives.html (there's a short description at https://salt.tips/the-coolest-features-in-salt-fluorine/#multiple-saltssh-versions)

**Comment by dmurphy18** [comment]
@HutTwoThreeFour Closing this issue as a 'will not fix', since UIbuntu 16.04 is long EOL, Python 3.5 is very EOL, oldest actively supported version of Python is Python 3.8 and then only CVEs.
With the reduced size of the Salt Core Team, dealing with very old EOL systems and software is not going to happen due to more pressing issues with currently supported systems.

**Comment by dmurphy18** [comment]
@HutTwoThreeFour You can try an older version of salt which supported Ubuntu 16.04, noting that that older version of Salt will have long been EOL too.  Another consideration to consider is whether that Ubuntu 16.04 is using Python 2 for a number of issues, and you could run into Python 2.7 vs Python 3.5 issues too, that is, things defaulting to Python 2.7 rather than Python 3.5.

I would use ssh directly to access the machine, given the age and how long Python and the OS have been EOL (End-Of-Lifed).
Lastly it was Salt 2018.x beforee Python 3 support was pretty much there in Salt.
